src/db/migrations/020_add_contact_trust_fields.sql

```
@@ -0,0 +1,16 @@
+-- trust_level: optional per-contact override for channel trust weight. NULL means use channel default.
+-- last_seen_at: timestamp of most recent inbound message from this contact.
+ALTER TABLE contacts ADD COLUMN contact_confidence NUMERIC(3,2) NOT NULL DEFAULT 0.0;
+ALTER TABLE contacts ADD COLUMN trust_level TEXT;
+ALTER TABLE contacts ADD COLUMN last_seen_at TIMESTAMPTZ;
+
+-- Index for finding high-confidence contacts (e.g. for UI display, future scoring queries)
+CREATE INDEX idx_contacts_confidence ON contacts (contact_confidence) WHERE contact_confidence > 0;
+
+-- Constraint: trust_level must be a valid value when set
+ALTER TABLE contacts ADD CONSTRAINT contacts_trust_level_check
+  CHECK (trust_level IN ('high', 'medium', 'low') OR trust_level IS NULL);
```

SQLFluff (4.1.0)

[error] Line 12, Position 71: Found unparsable section: 'WHERE contact_confidence > 0;\n\n-- Constr...'

(PRS)